pallets/msa/src/lib.rs

```diff
@@ -545,6 +545,16 @@ impl<T: Config> Pallet<T> {
 		Ok(())
 	}
 
+	/// Checks that the key is not already associated with another MSA
+	pub fn is_key_taken(msa_id: MessageSourceId, key: &T::AccountId) -> DispatchResult {
```

Comment:
I dont think we need this.

Reply:
I am ok with combining it somewhere, just want to make sue we differentiate between DuplicateKey and key which is already registered to another MSA

Reply:
@enddynayn  I combined the two in one where key is deleted from MSAKeys as well as KeyInfo

lets talk about its transactional nature

Reply:
@enddynayn and I discussed moving the check for `KeyAlreadyRegistered` from `is_key_taken` and placed directly into the `add_key` helper function